src/metabase/api/session.clj

```diff
@@ -360,7 +360,7 @@
   (api/let-404 [pulse-channel (t2/select-one PulseChannel :pulse_id pulse-id :channel_type "email")]
     (let [emails (get-in pulse-channel [:details :emails])]
       (if (some #{email} emails)
-        (t2/update! PulseChannel (:id pulse-channel) (assoc-in pulse-channel [:details :emails] (remove #{email} emails)))
+        (t2/update! PulseChannel (:id pulse-channel) (update-in pulse-channel [:details :emails] #(remove #{email} %)))
```

Comment:
I don't understand how this solves the race-condition. `pulse-channel` is immutable and so is `emails`, so `emails` before the cange is the same as the argument (`%` ) to the anon function after the change.

Reply:
The issue is that query value can be stale - the database could have been changed by a concurrent API call.

Have I missed something around transaction isolation - do API calls implicitly run inside a serializable transaction? I assume not for performance reasons.

Reply:
Ah nevermind, I see what you mean - I though that `pulse-channel` was a reference to the entity returned within the update transaction. I'll figure out how to do this properly with toucan.